services/core/shared/core/compat/sdk_v5/blockchainIndex.js

```diff
@@ -285,16 +292,20 @@ const cacheLegacyAccountInfo = async () => {
 
 const performGenesisAccountsIndexing = async () => {
 	const accountsDB = await getAccountsIndex();
-	const [genesisBlock] = await getBlockByHeight(await getGenesisHeight(), true);
+	const keyValueDB = await getKeyValueStoreIndex();
 
+	const [genesisBlock] = await getBlockByHeight(await getGenesisHeight(), true);
 	const genesisAccountsToIndex = genesisBlock.asset.accounts
 		.filter(account => account.address.length === 40)
 		.map(account => account.address);
-	const genesisAccountPageCached = 'genesisAccountPageCached';
 
 	logger.info(`${genesisAccountsToIndex.length} registered accounts found in the genesis block`);
 
-	const lastCachedPage = await genesisAccountsCache.get(genesisAccountPageCached) || 0;
+	const [{ value: lastCachedPageStr } = {}] = await keyValueDB.find(
+		{ key: genesisAccountPageCached, limit: 1 },
+		['value'],
+	);
```

Comment:
These `get/set` functions worked pretty well before. Considering we're switching to MySQL and treat one table as key-value store, can you please add a little bit abstraction here? I mean put that logic into the get/set functions and we can consider it ready.